src/Missings.jl

```diff
@@ -514,4 +514,73 @@ julia> emptymissing(first)([1], 2)
 """
 emptymissing(f) = (x, args...; kwargs...) -> isempty(x) ? missing : f(x, args...; kwargs...)
 
-end # module
+# Only for internal use. Allows dispatch over anonymous functions.
+struct MissingSmallest{T}
+    lt::T
+end
+
+"""
+    missingsmallest(f)
+
+Creates a function of two arguments `x` and `y` that tests whether `x` is less
+than `y` such that `missing` is always less than the other argument. In other 
+words, returns a modified version of the partial order function `f` such that
+`missing` is the smallest possible value, and all other non-`missing` values are
+compared according to `f`.
+
+The traditional `isless` function modified to treat `missing` as the smallest
+value can be accessed by using the 2-argument `missingsmallest(x, y)` function,
+see [`missingsmallest`](@ref).
+
+# Examples
+```
+julia> ismuchless = missingsmallest((x, y) -> isless(x, 10*y))
```

Comment:
`ismuchless` is not a valid comparison function for `sort!`, I'd rather use an example that is a valid comparison function for `sort!`.

Reply:
Sorry, I don't see why ismuchless is not a valid comparison function. It works like this when you use it with sort

julia> v = [missing, 10, missing, 1, 2]
julia> sort(v, lt=ismuchless)
5-element Vector{Union{Missing, Int64}}:
   missing
   missing
  2
  1
 10
The 2 was placed before 1, but that makes intuitive sense right?

Reply:
In Julia 1.9

```
julia> sort(1:1000, lt=(x, y) -> isless(x, 10*y))
1000-element Vector{Int64}:
 1000
  999
  998
  997
  996
  995
    ⋮
    5
    4
    3
    2
    1
```

In Julia 1.6

```
julia> sort(collect(1:10000), lt=(x, y) -> isless(x, 10*y))

signal (11): Segmentation fault: 11
in expression starting at REPL[6]:1
getindex at ./array.jl:805 [inlined]
partition! at ./sort.jl:554
sort! at ./sort.jl:570
sort! at ./sort.jl:659 [inlined]
#sort!#8 at ./sort.jl:719 [inlined]
sort!##kw at ./sort.jl:707 [inlined]
#sort#9 at ./sort.jl:766 [inlined]
sort##kw at ./sort.jl:766
jl_apply_generic at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
do_call at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
eval_body at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
jl_interpret_toplevel_thunk at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
jl_toplevel_eval_flex at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
jl_toplevel_eval_flex at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
jl_toplevel_eval_in at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
eval at ./boot.jl:360 [inlined]
eval_user_input at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:139
repl_backend_loop at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:200
start_repl_backend at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:185
#run_repl#42 at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:317
run_repl at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:305
jl_apply_generic at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
#881 at ./client.jl:387
jfptr_YY.881_31907 at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/sys.dylib (unknown line)
jl_apply_generic at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
jl_f__call_latest at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
#invokelatest#2 at ./essentials.jl:708 [inlined]
invokelatest at ./essentials.jl:706 [inlined]
run_main_repl at ./client.jl:372
exec_options at ./client.jl:302
_start at ./client.jl:485
jfptr__start_43472 at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/sys.dylib (unknown line)
jl_apply_generic at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
true_main at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
repl_entrypoint at /Users/x/.julia/juliaup/julia-1.6.7+0.x64.apple.darwin14/lib/julia/libjulia-internal.1.6.dylib (unknown line)
Allocations: 4050783 (Pool: 4049777; Big: 1006); GC: 5
```

An excerpt from the 1.11 docstring: lt must be "transitive in equivalence: !lt(x, y) && !lt(y, x) and !lt(y, z) && !lt(z, y) together imply !lt(x, z) && !lt(z, x). In words: if x and y are equivalent and y and are equivalent then x and z must be equivalent."

Reply:
Yes - I would recommend some natural example, e.g. comparing strings by their length (and if they have the same length they are equivalent).

Reply:
Wow, I had no idea that could cause such a serious error 🫠 Today I learned
I will remove the example entirely then